### ISSUE TITLE
summary() broken with --reconfigure --debug and a bool variable

### ISSUE THREAD

**Issue by nh2** [mesonbuild/meson#6646]
**Describe the bug**

`meson.build`:

```meson
summary({'asdf': true})
```

Run:

```
rm -rf build && meson setup build && meson setup build --debug --reconfigure
```

The first `setup` works, the `setup --debug --reconfigure` breaks, as shown in this output:

```
serkezet 1.0

    asdf: True

Found ninja-1.8.2 at /nix/store/v3rwd0gqjzvr8zppxy3w0nli2q7n0j54-ninja-1.8.2/bin/ninja
Traceback (most recent call last):
  File "/nix/store/xcar77agh5khbzcr1npmac56qbk1wgxy-meson-0.53.1/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 129, in run
    return options.run_func(options)
  File "/nix/store/xcar77agh5khbzcr1npmac56qbk1wgxy-meson-0.53.1/lib/python3.6/site-packages/mesonbuild/msetup.py", line 245, in run
    app.generate()
  File "/nix/store/xcar77agh5khbzcr1npmac56qbk1wgxy-meson-0.53.1/lib/python3.6/site-packages/mesonbuild/msetup.py", line 159, in generate
    self._generate(env)
  File "/nix/store/xcar77agh5khbzcr1npmac56qbk1wgxy-meson-0.53.1/lib/python3.6/site-packages/mesonbuild/msetup.py", line 220, in _generate
    coredata.update_cmd_line_file(self.build_dir, self.options)
  File "/nix/store/xcar77agh5khbzcr1npmac56qbk1wgxy-meson-0.53.1/lib/python3.6/site-packages/mesonbuild/coredata.py", line 836, in update_cmd_line_file
    config['options'].update(options.cmd_line_options)
  File "/nix/store/i1hh6prcyfqd066jdh4zwqfs58mqhdfw-python3-3.6.6/lib/python3.6/_collections_abc.py", line 841, in update
    self[key] = other[key]
  File "/nix/store/i1hh6prcyfqd066jdh4zwqfs58mqhdfw-python3-3.6.6/lib/python3.6/configparser.py", line 1237, in __setitem__
    self._parser._validate_value_types(option=key, value=value)
  File "/nix/store/i1hh6prcyfqd066jdh4zwqfs58mqhdfw-python3-3.6.6/lib/python3.6/configparser.py", line 1177, in _validate_value_types
    raise TypeError("option values must be strings")
TypeError: option values must be strings
```

Doesn't happen without `--debug`.

CC @xclaesse as the author of `summary()` (not sure if the bug is in `summary()` or somewhere more generic).

**Comment by nh2** [comment]
Workaround I'm using for now:

```meson
summary({'asdf': '@0@'.format(true)})
```

**Comment by xclaesse** [comment]
That's not related to summary(), it crash in the code that remember the command line options you gave. I think it would crash when passing any boolean built-in option, --debug here.

**Comment by xclaesse** [comment]
@nh2 it should be fixed with https://github.com/mesonbuild/meson/pull/6647, if you want to give it a try.

**Comment by nh2** [comment]
> @nh2 it should be fixed with #6647, if you want to give it a try.

Done (I commented, it works).
